### ISSUE TITLE
Cancel Print Behaviour

### ISSUE THREAD

**Issue by jrpmedia** [foosel/OctoPrint#1399]
Please read the "guidelines for contributing" that are linked ^-- just
up there. Also read the FAQ: https://github.com/foosel/OctoPrint/wiki/FAQ.

This is a bug and feature tracker, please only use it to report bugs
or request features within OctoPrint (not OctoPi, not any OctoPrint
plugins and not unofficial OctoPrint versions). Mark requests with
a [Request] prefix in the title please. Fully fill out the bug reporting
template for bug reports.

Do not seek support here ("I need help with ..."), that belongs on
the mailing list or the G+ community (both linked in the "guidelines
for contributing" linked above, read it!), NOT here.

Thank you!

----

#### What were you doing?

Cancelling a Print

#### What did you expect to happen?

The Machine to Stop printing ... and then Park itself to Home Axis.

#### What happened instead?

The Machine just Stops.
With the Nozzles Heads still on the Bed.
In other words it is a Dead Stop .. not so much a Cancel print.

#### Branch & Commit or Version of OctoPrint

OCTO Print v1.1.23 (master branch)

#### Printer model & used firmware incl. version

Flash Forge Creator Pro DUAL 2014
Sail Dish v7.7 r01234
Thing 32084 14/11/05

#### Browser and Version of Browser, Operating System running Browser

Chrome Version 52.0.2743.60 beta (64-bit)
Mac OSX El Capitam latest

#### Link to octoprint.log

https://dl.dropboxusercontent.com/u/5485939/OCTOPRINT/octoprint%20%282%29.log

#### Link to contents of terminal tab or serial.log

where do I get these from? :-(

#### Link to contents of Javascript console in the browser

where do I get these from? :-(

#### Screenshot(s) showing the problem:

n/a

I have read the FAQ.

**Comment by foosel** [comment]
That is stock cancel behaviour and intentional. While you might expect a cancel to also include a home, other people would tear my head off if I made that the default because "how dare you to just make my printer move!". That is why this is all configurable. You can adjust the script that is run on cancelling a print under "Settings" > "GCODE Scripts" > "After print job is cancelled".

By default it is this:

```
; disable motors
M84

;disable all heaters
{% snippet 'disable_hotends' %}
{% snippet 'disable_bed' %}
;disable fan
M106 S0
```

If you want to also home the printer on X and Y (I'd strongly advise against Z ;)), just append `G28 X0 Y0`:

```
; disable motors
M84

;disable all heaters
{% snippet 'disable_hotends' %}
{% snippet 'disable_bed' %}
;disable fan
M106 S0

; home x and y axis
G28 X0 Y0
```

> where do I get these from? :-(

It's all in the [Contribution Guidelines](https://github.com/foosel/OctoPrint/blob/master/CONTRIBUTING.md#how-to-file-a-bug-report) which you were asked to please read first on the very first line of the ticket input and in a big yellow box on top of the ticket form ;)
